### PR TITLE
init cue dsl function

### PR DESCRIPTION
**Pull request by wonderflow**
use cue template instead of parameters

### CODE REVIEW

File: pkg/utils/system/system.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/cloud-native-application/rudrx/api/types"
+)
+
+const RudrHome = ".rudr"
```
Comment (hongchaodeng) [inline pkg/utils/system/system.go:12]: rudr -> oam?

File: pkg/utils/system/system.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+func GetHomeDir() (string, error) {
```
Comment (hongchaodeng) [inline pkg/utils/system/system.go:14]: GetRudrHomeDir()

File: pkg/utils/system/system.go (hunk 1 above)
Comment (hongchaodeng) [inline pkg/utils/system/system.go:12]: This is not used externally. Should make it private.

File: pkg/cue/testdata/apps/myapp.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        ports:
+          http:
+            service_port: 80
+            container_port: '${echo_server_port}'
+    scopes:
+      - public-scope
+secrets:
+  secret-foo:
+    key1: '{{ config1 }}'
+    key2: '{{ config2 }'
```
Comment (hongchaodeng) [inline pkg/cue/testdata/apps/myapp.yaml:27]:
```suggestion
    key2: '{{ config2 }}'
```

File: pkg/cue/testdata/apps/myapp.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: myapp
```
Comment (hongchaodeng) [inline pkg/cue/testdata/apps/myapp.yaml:1]: Where is this yaml used?

File: config/samples/workloads/deployment_definition.yaml [hunk 5]
```diff
@@ -7,29 +7,31 @@ spec:
...
-        type: string
-        fieldPaths:
-          - "spec.containers[0].image"
-      - name: port
-        short: p
-        required: false
-        type: int
-        fieldPaths:
-          - "spec.containers[0].ports[0].containerPort"
+    template: |
```
Comment (hongchaodeng) [inline config/samples/workloads/deployment_definition.yaml:10]: It is better to put cue template into a separate file.
We can use cli to specify the .cue file for a definition.

File: pkg/cue/testdata/apps/myapp.yaml (hunk 4 above)
Comment (wonderflow) [reply pkg/cue/testdata/apps/myapp.yaml:1]: just example now

File: config/samples/workloads/deployment_definition.yaml (hunk 5 above)
Comment (wonderflow) [reply config/samples/workloads/deployment_definition.yaml:10]: next PR

File: pkg/utils/system/system.go (hunk 1 above)
Comment (wonderflow) [reply pkg/utils/system/system.go:12]: agree

File: pkg/utils/system/system.go (hunk 2 above)
Comment (wonderflow) [reply pkg/utils/system/system.go:14]: agree

**Pull request merged**